src/frontends/tensorflow_common/src/op/cumsum.cpp

```diff
@@ -14,9 +18,30 @@ namespace tensorflow {
 namespace op {
 
 OutputVector translate_cumsum_op(const NodeContext& node) {
-    default_op_checks(node, 2, {"Cumsum"});
+    default_op_checks(node, 2, {"Cumsum"}, true);
+
     auto x = node.get_input(0);
     auto axis = node.get_input(1);
+
+    auto complex_type_mark = as_type_ptr<ComplexTypeMark>(x.get_node_shared_ptr());
+    if (complex_type_mark) {
+        element::Type complex_part_type = complex_type_mark->get_complex_part_type();
+        x = complex_type_mark->input_value(0);
+
+        OutputVector concat_inputs;
+        concat_inputs.push_back(axis);
+        concat_inputs.push_back(make_shared<v0::Constant>(axis.get_element_type(), Shape{1}, 2));
+
+        auto concat = make_shared<v0::Concat>(concat_inputs, 0);
+        auto reshape = make_shared<v1::Reshape>(x, concat, false);
+        set_node_name(node.get_name(), reshape);
+        auto complex_reshape = make_shared<ComplexTypeMark>(reshape, complex_part_type);
+        return {complex_reshape->output(0)};
+    }
+
+    auto reshape = make_shared<v1::Reshape>(x, axis, false);
+    set_node_name(node.get_name(), reshape);
+    return {reshape};
```

Comment:
the implementation is incorrect. Please read instructions twice how to handle complex type